web/src/routes/admin/contests/import/+page.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import FormAlert from '$lib/FormAlert.svelte';
+	import type { Actions } from './$types';
+	import type { ContestImportData } from './+page.server';
+
+	export let form: Actions;
+
+	$: if (form && form.success) {
+		goto('/admin/contests');
+	}
+
+	let jsonText = '';
+	let parsesCorrectly: boolean | null = null;
+
+	let numProblems: number | null = null;
+	let numTeams: number | null = null;
+	let numSubmissions: number | null = null;
+
+	$: jsonText, updateUIFromJson();
+
+	function updateUIFromJson() {
+		try {
+			JSON.parse(jsonText);
+			const parsedContest: ContestImportData = JSON.parse(jsonText);
+			numProblems = parsedContest.Problems?.length ?? null;
+			numTeams = parsedContest.Teams?.length ?? null;
+			numSubmissions = parsedContest.Submissions?.length ?? null;
+			parsesCorrectly = numProblems > 0 && numTeams > 0 && parsedContest.Name?.length > 0;
+		} catch {
+			numProblems = null;
+			numTeams = null;
+			numSubmissions = null;
+			parsesCorrectly = false;
+		}
+
+		if (jsonText.length == 0) {
+			parsesCorrectly = null;
+		}
+	}
+</script>
+
+<svelte:head>
+	<title>Import Contest</title>
+</svelte:head>
+
+<h1 style="text-align:center" class="mb-4"><i class="bi bi-flag"></i> Import Contest</h1>
+
+<FormAlert />
+
+<form method="POST">
```

Comment:
Couldn't get `enhance` to work with this form for some reason, `await request.formData();` would hang forever, so I eventually gave up and this approach seems fine. If you want to dig into it, I can explain more.

Reply:
I tried `use:enhance` for me and it worked, so I just added it in. It worked for both failure case and successful case